lib/active_merchant/billing/gateways/credorax.rb

```diff
@@ -290,6 +291,8 @@ def add_payment_method(post, payment_method)
         post[:b5] = payment_method.verification_value
         post[:b4] = format(payment_method.year, :two_digits)
         post[:b3] = format(payment_method.month, :two_digits)
+        post[:token_eci] = payment_method&.eci if payment_method.is_a?(NetworkTokenizationCreditCard) && payment_method.source.to_s == 'network_token'
+        post[:token_crypto] = payment_method&.payment_cryptogram if payment_method.is_a?(NetworkTokenizationCreditCard) && payment_method.source.to_s == 'network_token'
```

Comment:
💭  Opinion:
I think it could be better if try something like this:
```
if payment_method.is_a?(NetworkTokenizationCreditCard)
  post[:token_eci] = payment_method&.eci if payment_method.source.to_s
  post[:token_crypto] = payment_method&.payment_cryptogram if payment_method.source.to_s == 'network_token'
end

  ```

Reply:
updated with a different approach